components/timezone/src/ixdtf.rs

```diff
@@ -78,116 +78,214 @@ impl UtcOffset {
-        date: &Option<DateRecord>,
-        time: &Option<TimeRecord>,
+impl TimeZoneInfo<models::AtTime> {
+    fn try_location_only_from_ixdtf_record(
+        ixdtf_record: &IxdtfParseRecord,
     ) -> Result<Self, ParseError> {
-        match record {
-            TimeZoneRecord::Name(iana_identifier) => {
-                let mapper = TimeZoneIdMapper::new();
-                let time_zone_id = mapper.as_borrowed().iana_bytes_to_bcp47(iana_identifier);
-
-                let mut zone_variant = None;
-                let mut local_time = None;
-                if let (Some(date), Some(time)) = (date, time) {
-                    let iso = DateTime::<Iso>::try_new_iso_datetime(
-                        date.year,
-                        date.month,
-                        date.day,
-                        time.hour,
-                        time.minute,
-                        time.second,
-                    )?;
-                    if let Some(offset) = offset {
-                        zone_variant = Some(
-                            if let Some((std_offset, dst_offset)) = ZoneOffsetCalculator::new()
-                                .compute_offsets_from_time_zone(time_zone_id, &iso)
-                            {
-                                if offset == std_offset {
-                                    ZoneVariant::standard()
-                                } else if Some(offset) == dst_offset {
-                                    ZoneVariant::daylight()
-                                } else {
-                                    return Err(ParseError::InvalidOffsetError);
-                                }
-                            } else {
-                                debug_assert_eq!(time_zone_id.0.as_str(), "unk");
-                                ZoneVariant::standard()
-                            },
-                        );
+        let IxdtfParseRecord {
+            offset: None,
+            tz:
+                Some(TimeZoneAnnotation {
+                    tz: TimeZoneRecord::Name(iana_identifier),
+                    ..
+                }),
+            date: Some(date),
+            time: Some(time),
+            ..
+        } = ixdtf_record
+        else {
+            return Err(ParseError::MismatchedTimeZoneFields);
+        };
+        let mapper = TimeZoneIdMapper::new();
+        let time_zone_id = mapper.as_borrowed().iana_bytes_to_bcp47(iana_identifier);
+        let iso = DateTime::<Iso>::try_new_iso_datetime(
+            date.year,
+            date.month,
+            date.day,
+            time.hour,
+            time.minute,
+            time.second,
+        )?;
+        Ok(Self {
+            time_zone_id,
+            offset: None,
+            zone_variant: (),
+            local_time: (iso.date, iso.time),
+        })
+    }
+}
+
+impl TimeZoneInfo<models::Full> {
+    fn try_full_from_ixdtf_record(ixdtf_record: &IxdtfParseRecord) -> Result<Self, ParseError> {
```

Comment:
nit: keep constructors for the same type together

Reply:
Right now all the `TimeZoneInfo` constructors are together and the `CustomZonedDateTime` constructors are together.